flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/tasks/StreamTask.java

```diff
@@ -877,12 +875,6 @@ public StreamStatusMaintainer getStreamStatusMaintainer() {
         CompletableFuture<Boolean> result = new CompletableFuture<>();
         mainMailboxExecutor.execute(
                 () -> {
-                    latestAsyncCheckpointStartDelayNanos =
```

Comment:
Why is this moved to the "inner" method? I found it a bit confusing that the "non-async" method not determines the start of the async part.

Reply:
This variable is only used for the true source tasks which do not attached with a `CheckpointBarrierHandler`, thus initially I think it could be moved to the base class for the source tasks would make it more clear. 

The variable mainly measures the delay between JM triggering checkpoint and (source) tasks start to processing this checkpoint (namely `checkpointStartDelayNanos`), since move it to be inside `triggerCheckpoint` would only differs in the time of one method call, thus I think it would not cause too large difference, and we could rename the variable to `latestACheckpointStartDelayNanos`. Do you think this option would also be ok?